Helper/AnalyticsHelper.php

```diff
@@ -0,0 +1,380 @@
+    }
+
+    public function getTopFiltersForASearch($search, array $params)
+    {
+        return $this->fetch(self::ANALYTICS_FILTER_PATH . '?search=' . urlencode($search), $params);
+    }
+
+    public function getTopFiltersForAttributesAndSearch(array $attributes, $search, array $params)
+    {
+        return $this->fetch(self::ANALYTICS_FILTER_PATH . '/' . implode(',', $attributes)
+            . '?search=' . urlencode($search), $params);
+    }
+
+    public function getTopFiltersForAttribute($attribute, array $params)
+    {
+        return $this->fetch(self::ANALYTICS_FILTER_PATH . '/' . $attribute, $params);
+    }
+
+    /**
+     * Click Analytics
+     *
+     * @param array $params
+     * @return mixed
+     */
+    public function getAverageClickPosition(array $params)
+    {
+        if (!$this->clickPositions) {
+            $this->clickPositions = $this->fetch(self::ANALYTICS_CLICKS_PATH . '/averageClickPosition', $params);
+        }
+
+        return $this->clickPositions;
+    }
+
+    public function getAverageClickPositionByDates(array $params)
+    {
+        $click = $this->getAverageClickPosition($params);
+
+        return $click && isset($click['dates']) ? $click['dates'] : [];
+    }
+
+    public function getClickThroughRate(array $params)
+    {
+        if (!$this->clickThroughs) {
+            $this->clickThroughs = $this->fetch(self::ANALYTICS_CLICKS_PATH . '/clickThroughRate', $params);
+        }
+
+        return $this->clickThroughs;
+    }
+
+    public function getClickThroughRateByDates(array $params)
+    {
+        $click = $this->getClickThroughRate($params);
+
+        return $click && isset($click['dates']) ? $click['dates'] : [];
+    }
+
+    public function getConversionRate(array $params)
+    {
+        if (!$this->conversions) {
+            $this->conversions = $this->fetch('/2/conversions/conversionRate', $params);
+        }
+
+        return $this->conversions;
+    }
+
+    public function getConversionRateByDates(array $params)
+    {
+        $conversion = $this->getConversionRate($params);
+
+        return $conversion && isset($conversion['dates']) ? $conversion['dates'] : [];
+    }
+
+    /**
+     * Client Data Check
+     *
+     * @return mixed
+     */
+    public function getClientData()
+    {
+        if (!$this->clientData) {
+            $this->clientData = $this->getClientSettings();
+        }
+
+        return $this->clientData;
+    }
+
+    public function isAnalyticsApiEnabled()
+    {
+        $clientData = $this->getClientData();
+
+        return $clientData && isset($clientData['analytics_api']) ? $clientData['analytics_api'] : 0;
+    }
+
+    public function isClickAnalyticsEnabled()
+    {
+        if (!$this->configHelper->isClickConversionAnalyticsEnabled()) {
+            return false;
+        }
+
+        $clientData = $this->getClientData();
+
+        return $clientData && isset($clientData['click_analytics']) ? $clientData['click_analytics'] : 0;
```

Comment:
Can you return `bool` value from here? Now the method might return `bool` / `int` / `string` and it's a good practice to strictly return only one type out of a method.